### PR TITLE
Edit Developer Guide Glossary

### LINKED ISSUE #52: Update Glossary: Define terms that are worth recording.

### PR DESCRIPTION
**Pull request by nvmdava**
Closes #52

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -304,11 +304,32 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-*{More to be added}*
+**Use case: Delete a Customer**
+
+**MSS**
+
+1.  User requests to list customers
+2.  ArtBuddy shows a list of customers
+3.  User requests to delete a customer in the list
+4.  ArtBuddy deletes the customer
```
Comment (chloeelim) [inline docs/DeveloperGuide.md:318]:
```suggestion
4.  ArtBuddy warns the user that all commissions under the customer will be deleted, and confirms whether the user still wants to proceed
5.  User confirms.
6.  ArtBuddy deletes the customer
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -304,11 +304,32 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. ArtBuddy shows an error message.
+
+      Use case resumes at step 2.
```
Comment (chloeelim) [inline docs/DeveloperGuide.md:333]:
```suggestion

* 4a. The user chooses not to proceed with the deletion.

  Use case ends.

```

**Review by chloeelim — COMMENTED** [review]
Hm, not very sure about this- but deleting customer should also delete their commission to prevent orphaning, so should the deletion of all commissions tied to the customer also be in the MSS?

**Comment by nvmdava** [comment]
Yeah, I think that works, thanks for the suggestion.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -321,8 +348,9 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
-* **Private contact detail**: A contact detail that is not meant to be shared with others
-
+* **Customer**: A contact detail, contains information about the customer, and a list of Commissions.
+* **Commission**: A single commission made by a customer, contains specifics about the commission and list of Iterations.
+* **Iteration**: A single iteration of the commission, contains an image for the commission and a text comment on the image.
```
Comment (chloeelim) [inline docs/DeveloperGuide.md:353]:
```suggestion
* **Customer**: A contact detail. Contains information about the customer, and a list of commissions.
* **Commission**: An art piece requested by a customer. Contains specifics about the commission and list of iterations.
* **Iteration**: A single version of a commission. Contains an image and a text comment on the image. 
```

**Review by chloeelim — COMMENTED** [review]
The definitions for the glossary seem too 'circular', if they are self explanatory I think they shouldn't be in the glossary. In particular, I think `commission` could be explained in simpler terms because non-artsy people (ie: most developers) might not be familiar with a `commission` in context of arts.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -321,8 +348,9 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
-* **Private contact detail**: A contact detail that is not meant to be shared with others
-
+* **Customer**: A contact detail. Contains information about the customer, and a list of commissions.
+* **Commission**: An art piece requested by a customer that has been delivered or under progress. Contains specifics about the commission and list of iterations.
```
Comment (ngxingyu) [inline docs/DeveloperGuide.md:352]:
```suggestion
* **Commission**: An art piece requested by a customer that has been delivered or is in progress. Contains specifics about the commission and a list of iterations.
```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -285,14 +285,14 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to list customers
+2.  ArtBuddy shows a list of customers
+3.  User requests to open customer in the list
+4.  ArtBuddy shows the list of commissions the customer made and the detail of the customer
```
Comment (ngxingyu) [inline docs/DeveloperGuide.md:295]:
```suggestion
4.  ArtBuddy shows the list of commissions the customer made and the details of the customer
```

**Review by ngxingyu — CHANGES_REQUESTED** [review]
some minor grammar fixes, I'm fine with the rest of it.

**Review by ngxingyu — APPROVED** [review]
